File: backend/candidate-profile-service/src/main/java/com/jobportal/candidateprofile/controllers/CandidateSkillsController.java
package com.jobportal.candidateprofile.controllers;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import com.jobportal.candidateprofile.dto.CandidateSkillsDto;
import com.jobportal.candidateprofile.service.CandidateSkillsService;

@RestController
@RequestMapping("/api/candidate-skills")
@CrossOrigin("*")
public class CandidateSkillsController {

    @Autowired
    private CandidateSkillsService service;

    @PostMapping
    public CandidateSkillsDto addSkill(@RequestBody CandidateSkillsDto dto) {
        return service.addSkill(dto);
    }

    @PutMapping("/{csId}")
    public CandidateSkillsDto updateSkill(@PathVariable Integer csId, @RequestBody CandidateSkillsDto dto) {
        return service.updateSkill(csId, dto);
    }

    @DeleteMapping("/{csId}")
    public String deleteSkill(@PathVariable Integer csId) {
        service.deleteSkill(csId);
        return "Candidate skill deleted successfully";
    }

    @GetMapping("/{csId}")
    public CandidateSkillsDto getSkillById(@PathVariable Integer csId) {
        return service.getSkillById(csId);
    }

    @GetMapping("/candidate/{cid}")
    public List<CandidateSkillsDto> getSkillsByCandidate(@PathVariable Integer cid) {
        return service.getSkillsByCandidate(cid);
    }

    @PutMapping("/candidate/{cid}")
    public List<CandidateSkillsDto> updateSkillsByCandidate(@PathVariable Integer cid, @RequestBody List<String> skillNames) {
        return service.saveSkillsForCandidate(cid, skillNames);
    }

    @GetMapping
    public List<CandidateSkillsDto> getAllSkills() {
        return service.getAllSkills();
    }
}

File: backend-dotnet/Services/JobPortal.CandidateService/Controllers/CandidateSkillsController.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using JobPortal.CandidateService.DTOs;
using JobPortal.CandidateService.Services;

namespace JobPortal.CandidateService.Controllers;

[ApiController]
[Route("api/candidate-skills")]
public class CandidateSkillsController : ControllerBase
{
    private readonly CandidateSkillsService _service;

    public CandidateSkillsController(CandidateSkillsService service)
    {
        _service = service;
    }

    [HttpPost]
    public async Task<IActionResult> AddSkill([FromBody] CandidateSkillsDto dto)
    {
        var result = await _service.AddSkillAsync(dto);
        return Ok(result);
    }

    [HttpPut("{csId}")]
    public async Task<IActionResult> UpdateSkill(int csId, [FromBody] CandidateSkillsDto dto)
    {
        try
        {
            var result = await _service.UpdateSkillAsync(csId, dto);
            return Ok(result);
        }
        catch (KeyNotFoundException)
        {
            return NotFound("Candidate skill not found");
        }
    }

    [HttpDelete("{csId}")]
    public async Task<IActionResult> DeleteSkill(int csId)
    {
        try
        {
            await _service.DeleteSkillAsync(csId);
            return Ok("Candidate skill deleted successfully");
        }
        catch (KeyNotFoundException)
        {
            return NotFound("Candidate skill not found");
        }
    }

    [HttpGet("{csId}")]
    public async Task<IActionResult> GetSkillById(int csId)
    {
        try
        {
            var result = await _service.GetSkillByIdAsync(csId);
            return Ok(result);
        }
        catch (KeyNotFoundException)
        {
            return NotFound("Candidate skill not found");
        }
    }

    [HttpGet("candidate/{cid}")]
    public async Task<IActionResult> GetSkillsByCandidate(int cid)
    {
        var list = await _service.GetSkillsByCandidateAsync(cid);
        return Ok(list);
    }

    [HttpPut("candidate/{cid}")]
    public async Task<IActionResult> UpdateSkillsByCandidate(int cid, [FromBody] List<string> skillNames)
    {
        var result = await _service.SaveSkillsForCandidateAsync(cid, skillNames);
        return Ok(result);
    }

    [HttpGet]
    public async Task<IActionResult> GetAllSkills()
    {
        var list = await _service.GetAllSkillsAsync();
        return Ok(list);
    }
}
